docs/misc/math-expr.md

```diff
@@ -119,6 +119,13 @@ See javadoc of java.lang.Math for detailed explanation for each function.
 |acos|acos(x) would return the arc cosine of x|
 |asin|asin(x) would return the arc sine of x|
 |atan|atan(x) would return the arc tangent of x|
+|bitwiseAnd|bitwiseAnd(x,y) would return the result of x & y. Double values will be converted to their bit representation|
+|bitwiseComplement|bitwiseComplement(x) would return the result of ~x. Double values will be converted to their bit representation|
+|bitwiseConvertDouble|bitwiseConvertDouble(x) would convert the IEEE 754 floating-point "double" bits stored in a long into a double value if the input is a long, or the copy bits of a double value into a long if the input is a double.|
```

Comment:
This function is kind of weird because it doesn't have a fixpoint. I'd think that `bitwiseConvertDouble(bitwiseConvertDouble(x))` would be identical to `bitwiseConvertDouble(x)`. The lack of fixpoint makes it hard to reason about what the result of this function is going to be. Is there a specific reason it's designed this way? If not, I'd suggest splitting into two functions for each direction of the conversion.

Reply:
will split into `bitwiseConvertDoubleToLongBits` and `bitwiseConvertLongBitsToDouble`

Reply:
>I'd think that `bitwiseConvertDouble(bitwiseConvertDouble(x))` would be identical to `bitwiseConvertDouble(x)`.

hmm, should the conversion just pass through if the type is already the output type or should it implicitly cast similar to the other bitwise functions?

Reply:
It should implicitly cast, I think.

Generally I think function behavior is easier to understand if the function implicitly casts its inputs to the type that it expects, vs. changing behavior based on its input type.

Reply:
btw, testing `bitwiseConvertLongBitsToDouble(bitwiseConvertLongBitsToDouble(..))` uncovered an issue with the parser when trying to parse the output of `Expr.stringify` (because unit tests cover this round trip scenario, and when flatten is true it turns it into a constant), where large doubles with exponents, e.g. `1E10` or whatever, could not be correctly parsed, so I expanded the grammar to allow it [roughly according to these rules](https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html#valueOf-java.lang.String-)